### PR TITLE
CI/Replace Dockerhub to GitHub Container Registry

### PR DESCRIPTION
**Pull request by iGxnon**
Please briefly answer these questions:

Test workflow: https://github.com/iGxnon/Xline/actions/runs/5830042498

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Replace dockerhub to GitHub container registry  

* what changes does this pull request make?

    See above

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

    No

### CODE REVIEW

**Review by Phoenix500526 — DISMISSED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -76,19 +76,25 @@ jobs:
...
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          registry: ghcr.io
+          username: ${{ env.REPOSITORY }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Image ID
+        run: |
+          IMAGE_ID=ghcr.io/${{ env.REPOSITORY }}/xline
+          echo IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]') >> $GITHUB_ENV
```
Comment (themanforfree) [inline .github/workflows/docker.yml:89]: Why set up image ID at here, Could we set up it directly in `env`

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
 
 env:
   CI_RUST_TOOLCHAIN: 1.70.0
-  REGISTRY_IMAGE: datenlord/xline
+  REPOSITORY: ${{ github.actor }}
```
Comment (themanforfree) [inline .github/workflows/docker.yml:8]: The explanation of `github.actor` in GitHub Docs is as follows.
> The username of the user that triggered the initial workflow run.  

Should we hardcode it to `datenlord`?

Comment (iGxnon) [reply .github/workflows/docker.yml:8]: In order to test in my forked repository, I haven't hard-coded the `REPOSITORY`. None of these CIs will trigger in the PRs, so should the `github.actor` here always be `xline-kv`?

File: .github/workflows/docker.yml (hunk 1 above)
Comment (iGxnon) [reply .github/workflows/docker.yml:89]: If we avoid hard-coding the `REPOSITORY`, we must convert the image ID to lowercase before pushing it to GHCR.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (themanforfree) [reply .github/workflows/docker.yml:8]: This CI may be triggered manually, So the `github.actor` should be the username of the trigger.

Comment (iGxnon) [reply .github/workflows/docker.yml]: Fixed

**Review by themanforfree — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**